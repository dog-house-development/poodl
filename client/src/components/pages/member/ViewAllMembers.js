import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import { fetchMembers } from '../../../actions/memberActions';
import DataGrid from '../../ui/DataGrid';
import { Link } from 'react-router-dom';
import moment from 'moment';

export class ViewAllMembers extends Component {
    constructor(props) {
        super(props);
        this.handleRowClick = this.handleRowClick.bind(this);
    }

    componentDidMount() {
        // call redux action to retrieve all members from api
        this.props.getMembers();
    }

    handleRowClick(e, id) {
        e.preventDefault();
        this.props.history.push(`/members/${id}`);
    }

    getDataGridContent() {
        // choose what we want to display out of the members data
        let data = [];
        _.each(this.props.members, member => {
            // we want members' names, emails and membership dates.
            data.push({
                key: member._id,
                firstName: member.firstName,
                lastName: member.lastName,
                membershipDate: moment(member.membershipDate).format('MMMM Do, YYYY'),
                email: member.email
            });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container">
                <Link to="/dashboard" className="button small tertiary">
                    <i className="material-icons">keyboard_backspace</i> Back to home
                </Link>
                <h1>Manage Members</h1>
                <DataGrid
                    data={this.getDataGridContent()}
                    loading={this.props.loading}
                    onRowClick={this.handleRowClick}
                />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        members: state.members.all,
        loading: state.members.loading,
        errors: state.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        getMembers: () => dispatch(fetchMembers())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAllMembers);
