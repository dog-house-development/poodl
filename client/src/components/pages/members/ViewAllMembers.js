import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import _ from 'lodash';
import MemberActions from '../../../actions/memberActions';
import DataGrid from '../../ui/DataGrid';
import { Link } from 'react-router-dom';
import moment from 'moment';

export class ViewAllMembers extends Component {
    constructor(props) {
        super(props);
        this.handleRowClick = this.handleRowClick.bind(this);
    }

    componentDidMount() {
        // call redux action to retrieve all members from api
        this.props.memberActions.filter();
    }

    handleRowClick(e, id) {
        e.preventDefault();
        this.props.history.push(`/members/${id}`);
    }

    getDataGridContent() {
        // choose what we want to display out of the members data
        let data = [];
        _.each(this.props.members, member => {
            // we want members' names, emails and membership dates.
            data.push({
                key: member._id,
                firstName: member.firstName,
                lastName: member.lastName,
                membershipDate: moment(member.membershipDate).format('MMMM Do, YYYY'),
                email: member.email
            });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container page-container">
                <Link to="/dashboard" className="button small tertiary">
                    <i className="material-icons">keyboard_backspace</i> Back to home
                </Link>
                <div className="view-all-header">
                    <h1>Manage Members</h1>
                    <Link to="/members/register" className="button small primary">
                        <i className="material-icons button-icon">add</i>
                        Register Member
                    </Link>
                    <Link to="/member-check-in" className="button small primary">
                        <i className="material-icons button-icon">check</i>
                        Member check-in
                    </Link>
                </div>
                <DataGrid
                    data={this.getDataGridContent()}
                    loading={this.props.loading}
                    onRowClick={this.handleRowClick}
                />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        members: state.members.all,
        loading: state.members.loading,
        errors: state.members.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        memberActions: bindActionCreators(MemberActions, dispatch)
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAllMembers);
