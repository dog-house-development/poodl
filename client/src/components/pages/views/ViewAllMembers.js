import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import { fetchMembers } from '../../../actions/memberActions';
import DataGrid from '../../ui/DataGrid';
import { Link } from 'react-router-dom';

export class ViewAllMembers extends Component {
    onViewClick = (e, memberID) => {
        e.preventDefault();
        this.props.history.push('/memberProfile/:id', memberID);
    };
    componentDidMount() {
        // call redux action to retrieve all members from api
        this.props.getMembers();
    }

    getDataGridContent() {
        // choose what we want to display out of the members data
        let data = [];
        _.each(this.props.members, member => {
            // we want members' names, emails and membership dates.
            data.push({
                key: member._id,
                firstName: member.firstName,
                lastName: member.lastName,
                membershipDate: member.membershipDate,
                email: member.email,
                //on click needs
                viewProfile: (
                    <Link to={'/member/' + member._id} className="button large primary">
                        {' '}
                        View{' '}
                    </Link>
                )
            });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container">
                <h1>View All Members</h1>
                <DataGrid data={this.getDataGridContent()} loading={this.props.loading} />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        members: state.members.all,
        loading: state.members.loading,
        errors: state.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        getMembers: () => dispatch(fetchMembers())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAllMembers);
