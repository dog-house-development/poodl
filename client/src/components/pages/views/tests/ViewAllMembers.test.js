import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';
import { Link } from 'react-router-dom';
import { ViewAllMembers, mapStateToProps, mapDispatchToProps } from '../ViewAllMembers';

configure({ adapter: new Adapter() });

describe('ViewAllMembers tests', () => {
    let state, props, wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                members: {
                    loading: false,
                    all: [
                        {
                            _id: '123',
                            firstName: 'Big',
                            lastName: 'Tup',
                            membershipDate: '1950-03-15T20:07:42.762Z',
                            email: 'bigtup@nowhere.com'
                        },
                        {
                            _id: '321',
                            firstName: 'Lil',
                            lastName: 'Tup',
                            membershipDate: '1950-03-15T20:07:42.762Z',
                            email: 'liltup@nowhere.com'
                        }
                    ]
                },
                errors: {}
            },
            _state
        );
        props = _.assign({}, _props);
        wrapper = shallow(
            <ViewAllMembers
                {..._.assign(
                    {},
                    props,
                    mapStateToProps(state, props),
                    mapDispatchToProps(jasmine.createSpy('dispatch'))
                )}
            />
        );
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                members: [
                    {
                        _id: '123',
                        firstName: 'Big',
                        lastName: 'Tup',
                        membershipDate: '1950-03-15T20:07:42.762Z',
                        email: 'bigtup@nowhere.com'
                    },
                    {
                        _id: '321',
                        firstName: 'Lil',
                        lastName: 'Tup',
                        membershipDate: '1950-03-15T20:07:42.762Z',
                        email: 'liltup@nowhere.com'
                    }
                ],
                errors: {},
                loading: false
            });
        });
    });

    describe('mapDispatchToProps', () => {
        it('should map dispatch to props', () => {
            const dispatch = jest.fn();
            expect(JSON.stringify(mapDispatchToProps(dispatch))).toEqual(JSON.stringify({ getMembers: () => {} }));
        });
    });

    describe('componentDidMount', () => {
        it('should run without breaking', () => {
            spyOn(instance, 'componentDidMount');
            instance.componentDidMount();
        });
    });

    describe('getDataGridContent', () => {
        it('should return filtered members data', () => {
            expect(instance.getDataGridContent()).toEqual([
                {
                    key: '123',
                    firstName: 'Big',
                    lastName: 'Tup',
                    membershipDate: 'March 15th, 1950',
                    email: 'bigtup@nowhere.com'
                },
                {
                    key: '321',
                    firstName: 'Lil',
                    lastName: 'Tup',
                    membershipDate: 'March 15th, 1950',
                    email: 'liltup@nowhere.com'
                }
            ]);
        });
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
