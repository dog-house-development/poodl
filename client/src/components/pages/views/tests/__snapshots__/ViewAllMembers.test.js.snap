// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAllMembers tests render should render correctly 1`] = `
<div
  className="view-all-container"
>
  <Link
    className="button small tertiary"
    replace={false}
    to="/dashboard"
  >
    <i
      className="material-icons"
    >
      keyboard_backspace
    </i>
     Back to home
  </Link>
  <h1>
    View All Members
  </h1>
  <DataGrid
    data={
      Array [
        Object {
          "email": "bigtup@nowhere.com",
          "firstName": "Big",
          "key": "123",
          "lastName": "Tup",
          "membershipDate": "February 2nd, 2020",
          "viewProfile": <Link
            className="button medium primary"
            replace={false}
            to="/member/123"
          >
            View
          </Link>,
        },
        Object {
          "email": "liltup@nowhere.com",
          "firstName": "Lil",
          "key": "321",
          "lastName": "Tup",
          "membershipDate": "February 2nd, 2020",
          "viewProfile": <Link
            className="button medium primary"
            replace={false}
            to="/member/321"
          >
            View
          </Link>,
        },
      ]
    }
    loading={false}
  />
</div>
`;
